Cargo.toml

```diff
@@ -37,7 +37,10 @@ bitflags = "1.1.0"
 bson = { git = "https://github.com/mongodb/bson-rust" }
 chrono = "0.4.7"
 derivative = "2.1.1"
-futures = "0.3.5"
+futures-core = "0.3.14"
+futures-io = "0.3.14"
+futures-util = { version = "0.3.14", features = ["io"] }
+futures-executor = "0.3.14"
```

Comment:
The `futures` crate contains a number of async-related APIs that are widely used in the community, the most important one (for us) being the [`Stream`](https://docs.rs/futures-core/0.3.14/futures_core/stream/trait.Stream.html) trait, which our various cursor types conform to (it's basically the async iterator trait). Because we publicly conform to it, we'll need to do so for each major version of `futures` to ensure full compatibility. The `futures` crate is really just a collection of rexports from "sub" crates though (many of which are listed here now), each of which has differing stability exceptions. The `futures-core` one, which contains `Stream`, is purposefully much more stable than the others, and it will likely go straight from `0.3` to `1.0` without any more breaking versions, whereas `futures-util` and `futures` itself will have potentially many major versions. For that reason, I updated this to depend directly on `futures-core` instead of `futures` to reduce the number of separate `Stream` implementations we'll need to provide for the lifetime of 2.0. This [page](https://smallcultfollowing.com/babysteps/blog/2019/12/09/async-interview-2-cramertj/#the-futures-crate-is-really-a-facade) gives a nice overview of the situation and why we need to do this.